### PR TITLE
Add a timeout to catalog2lucene's index commits

### PR DESCRIPTION
**Pull request by loic-sharma**
Fixes https://github.com/NuGet/Engineering/issues/1838

Verify on DEV with:

```
traces
| where timestamp >= todatetime('2018-11-06T21:04:07.1397859Z') 
| where customDimensions.JobName == "catalog2lucene"
| order by timestamp asc
```

### CODE REVIEW

File: src/Ng/SearchIndexFromCatalogCollector.cs [hunk 1]
```diff
@@ -101,13 +105,48 @@ public void EnsureCommitted()
...
+                {
+                    commitTask = commitTask.TimeoutAfter(_commitTimeout.Value);
+                }
+
+                await commitTask;
+            }
+            catch (OperationCanceledException)
+            {
+                _telemetryService.TrackIndexCommitTimeout();
+                _logger.LogError("TIMEOUT Commit timed out!");
```
Comment (chenriksson) [inline src/Ng/SearchIndexFromCatalogCollector.cs]: Is there any other info we want to log here, like `NumDocs()`?

File: tests/NgTests/SearchIndexFromCatalogCollectorTests.cs [hunk 2]
```diff
@@ -102,6 +111,63 @@ public async Task DoesNotSkipPackagesWhenExceptionOccurs(string catalogUri, stri
...
+            await mockServer.AddStorageAsync(catalogStorage);
+
+            ReadWriteCursor front = new DurableCursor(
+                storage.ResolveUri("cursor.json"),
+                storage,
+                MemoryCursor.MinValue);
+            ReadCursor back = MemoryCursor.CreateMax();
+
+            var commitTimeout = TimeSpan.FromSeconds(1);
+            var stuckDuration = TimeSpan.FromMinutes(1);
```
Comment (chenriksson) [inline tests/NgTests/SearchIndexFromCatalogCollectorTests.cs]: Adds a minute to tests, or are they run in parallel? Could this be shorter?

Comment (loic-sharma) [reply tests/NgTests/SearchIndexFromCatalogCollectorTests.cs]: The timeout is set to a second, so this test completes after a second.

File: src/Ng/SearchIndexFromCatalogCollector.cs (hunk 1 above)
Comment (loic-sharma) [reply src/Ng/SearchIndexFromCatalogCollector.cs]: Good idea. Added

**Pull request merged**